// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

// option java_multiple_files = true;
// option java_package = "io.grpc.examples.helloworld";
// option java_outer_classname = "HelloWorldProto";
// option objc_class_prefix = "HLW";
option go_package="build/gen";

package smdbrpc;

message HLCTimestamp {
  int64 walltime = 1; // the 64 bit timestampe value
  int64 logicaltime = 2; // the causality counter
}

// The greeting service definition.
service HotshardGateway {
  // Sends a greeting
  rpc ContactHotshard (HotshardRequest) returns (HotshardReply) {}
}

// The request message containing the user's name.
message HotshardRequest {
  string sqlstring = 1;

  HLCTimestamp hlctimestamp = 2;  
}

// The response message containing the greetings
message HotshardReply {
  bool is_committed = 1;

  HLCTimestamp hlctimestamp = 2;
}
